startupscript/aws/post-startup-hook.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash -x
+
+# post-startup-hook.sh
+#
+# Performs additional one-time configuration for applications running in AWS
+# EC2 instances.
+#
+# Note that this script is intended to be sourced from the "post-startup.sh"
+# script and is dependent on some functions and variables already being set up
+# and some packages already installed:
+#
+# - aws (cli from ghcr.io/devcontainers/features/aws-cli:1)
+# - emit (function)
+# - CLOUD_SCRIPT_DIR: path where AWS-specific scripts live
+# - RUN_AS_LOGIN_USER: run command as app user
+# - USER_BASH_PROFILE: path to user's ~/.bash_profile file
+# - USER_BASHRC: path to user's ~/.bashrc file
+# - USER_NAME: name of app user
+# - USER_PRIMARY_GROUP: name of primary group app user belongs to
+# - USER_PROFILE: path to user's ~/.profile file
+# - USER_WORKBENCH_CONFIG_DIR: user's WB configuration directory
+# - WORKBENCH_GIT_REPOS_DIR: path to the git repo directory (~/repos)
+# - WORKBENCH_INSTALL_PATH: path to CLI executable
+
+readonly AWS_VAULT_INSTALL_PATH="/usr/bin/aws-vault"
+
+##########################################
+# Install aws-vault for credential caching
+##########################################
+emit "installing aws-vault"
+curl --no-progress-meter --location --output "${AWS_VAULT_INSTALL_PATH}" \
+    https://github.com/99designs/aws-vault/releases/download/v7.2.0/aws-vault-linux-amd64
```

Comment:
It's a nit, but I think this URL would be good as a variable defined up with the install path.